PR #1175 broken make build.parallel

We merged PR 1175 but discover that it brakes make build.parallel while make build is fine.

erwan@mr-meeseeks:~/ceph-container (master)$ sudo make FLAVORS=luminous,centos,7 build.parallel
make[1] : on entre dans le répertoire « /home/erwan/ceph-container »
make[2] : on entre dans le répertoire « /home/erwan/ceph-container/staging/luminous-centos-7-x86_64/daemon-base »
=== docker build ceph/daemon-base:master-luminous-centos-7-x86_64

  CEPH_VERSION      : luminous
  CEPH_POINT_RELEASE: 
  DISTRO            : centos
  DISTRO_VERSION    : 7
  HOST_ARCH         : x86_64
  BASEOS_REGISTRY   : 
  BASEOS_REPO       : centos
  BASEOS_TAG        : 7
  IMAGES_TO_BUILD   : daemon-base daemon
  STAGING_DIR       : staging/luminous-centos-7-x86_64
  RELEASE           : master
  DAEMON_BASE_IMAGE : ceph/daemon-base:master-luminous-centos-7-x86_64
  DAEMON_IMAGE      : ceph/daemon:master-luminous-centos-7-x86_64

Computed:
  BASE_IMAGE        : centos:7

make : on entre dans un répertoire inconnu
make: *** staging/luminous-centos-7-x86_64/daemon : Aucun fichier ou dossier de ce type. Arrêt.
make : on quitte un répertoire inconnu
make[1]: *** [daemon.x86_64,luminous,centos,7] Erreur 2
make[1]: *** Attente des tâches non terminées....
unable to prepare context: unable to get absolute context directory of given context directory ".": getwd: no such file or directory
make[2]: *** [build] Erreur 1
make[2] : on quitte le répertoire « /home/erwan/ceph-container/staging/luminous-centos-7-x86_64/daemon-base »
make[1]: *** [daemon-base.x86_64,luminous,centos,7] Erreur 2
make[1] : on quitte le répertoire « /home/erwan/ceph-container »
make: *** [build.parallel] Erreur 2

https://2.jenkins.ceph.com/job/ceph-container-prs-ceph_ansible-luminous-centos7-cluster/525/consoleText

@BlaineEXE We still have the same kind of issues as in 
https://2.jenkins.ceph.com/job/ceph-container-build-ceph-base-push-imgs/4/console
make[1]: *** staging/kraken-11.0.1-centos-7-x86_64/daemon: No such file or directory.  Stop.

Can you have a look on why the build.parallel fails ?

--- a/Makefile
+++ b/Makefile
@@ -56,20 +56,12 @@ ALL_BUILDABLE_FLAVORS := \
 stage.%:
 	@$(call set_env_vars,$*) sh -c maint-lib/stage.py
 
-# Target for daemon-base image; delegate build/clean/push goals to target makefile
-daemon-base.%:
-	$(call set_env_vars,$*); $(MAKE) -C $$STAGING_DIR/daemon-base \
-	  $(call set_env_vars,$*) $(filter-out daemon-base.$*,$(MAKECMDGOALS))
-
-# Target for daemon image; delegate build/clean/push goals to target makefile
-daemon.%:
-	@$(call set_env_vars,$*); $(MAKE) $(call set_env_vars,$*) -C $$STAGING_DIR/daemon \
-	  $(call set_env_vars,$*) $(filter-out daemon.$*,$(MAKECMDGOALS))
-
 # Make daemon-base.% and/or daemon.% target based on IMAGES_TO_BUILD setting
 #do.image.%: | stage.% $(foreach i, $(IMAGES_TO_BUILD), $(i).% ) ;
 do.image.%: stage.%
-	@set -eux ; for image in $(IMAGES_TO_BUILD); do $(MAKE) $${image}.$* build ; done
+	$(foreach i, $(IMAGES_TO_BUILD), \
+		$(call set_env_vars,$*); $(MAKE) $(call set_env_vars,$*) -C $$STAGING_DIR/$(i) \
+			$(call set_env_vars,$*) $(MAKECMDGOALS)$(\n))
 
 stage: $(foreach p, $(FLAVORS), stage.$(HOST_ARCH),$(p)) ;
 build: $(foreach p, $(FLAVORS), do.image.$(HOST_ARCH),$(p)) ;

--- a/maint-lib/makelib.mk
+++ b/maint-lib/makelib.mk
@@ -51,3 +51,10 @@ PARALLEL := --jobs $(nproc) --output-sync
 else
 PARALLEL := --jobs $(nproc)
 endif
+
+
+# define a newline
+define \n
+
+
+endef

